clickhouse-http-client/src/main/java/com/clickhouse/client/http/config/ClickHouseHttpOption.java

```diff
@@ -64,7 +65,22 @@ public enum ClickHouseHttpOption implements ClickHouseOption {
      * Only one role can be set at a time.
      */
     REMEMBER_LAST_SET_ROLES("remember_last_set_roles", false,
-            "Whether to remember last set role and send them in every next requests as query parameters.");
+            "Whether to remember last set role and send them in every next requests as query parameters."),
+
+    /**
+     * Whether to retry on failure with AsyncHttpClient. Failure includes some 'critical' IO exceptions:
+     * <ul>
+     *     <li>{@code org.apache.hc.core5.http.ConnectionClosedException}</li>
+     *     <li>{@code org.apache.hc.core5.http.NoHttpResponseException}</li>
+     * </ul>
+     *
+     * And next status codes:
+     * <ul>
+     *     <li>{@code 503 Service Unavailable}</li>
+     * </ul>
+     */
+    AHC_RETRY_ON_FAILURE("ahc_retry_on_failure", false, "Whether to retry on failure with AsyncHttpClient.")
```

Comment:
Can we change the name of the property?

Reply:
Yes, sure. But let me explain why I've chosen this one - implementation works only for Apache HC and only for old client. 
In the new client I would name it `http.retry_on_failure` meaning it will be supported for every http implementation we ever have.